networkdb/delegate.go

```diff
@@ -53,6 +53,7 @@ func (nDB *NetworkDB) handleNetworkEvent(nEvent *NetworkEvent) bool {
 			n.leaveTime = time.Now()
 		}
 
+		nDB.addNetworkNode(nEvent.NetworkID, nEvent.NodeName)
```

Comment:
If `n.leaving == true` shouldn't we set it to false now since the node has joined the network again ?